client/src/App.js

```diff
@@ -59,6 +60,11 @@ function App() {
         <PrivateRoute path="/checkout">
           <Checkout />
         </PrivateRoute>
+        <Route path="/404">
+          <PageNotFound />
+        </Route>
+        <Route exact path="/" component={Home} />
+        {/* <Redirect from="*" to="/404" /> */}
```

Comment:
```suggestion
        <Route component={PageNotFound} />
```

This route and wrapping all the routes in a Switch component worked for me!
https://ultimatecourses.com/blog/react-router-not-found-component

Reply:
I would recommend using a Switch component too. It renders the last item if no match is found. The caveat is that it only renders the first matching item as well but for our usage that shouldn't post an issue

Reply:
I considered that before but does a switch component conflict with the browser router?

Reply:
I don't believe so, they dhow an example of them both in the docs!
https://reactrouter.com/web/guides/quick-start